### ISSUE TITLE
Add 'with' variable instantiation detection to BadNameAttributeUseLinter

### ISSUE THREAD

**Issue by mschwager** [duo-labs/dlint#3]
I.e. [Context managers](https://docs.python.org/2.5/whatsnew/pep-343.html).

The `BadNameAttributeUseLinter` helper currently supports normal variable instantiation of an object, e.g.

```python
def func():
    foo = Foo()
    bar = foo.bad_function()  # Caught by BadNameAttributeUseLinter
    print(bar)
```

However, the helper does not support instantiation via the `with` statement, e.g.

```python
def func():
    with Foo() as foo:
        bar = foo.bad_function()  # Not caught by BadNameAttributeUseLinter, yet
        print(bar)
```

This pattern is not as common as normal variable instantiation, however, it is worth detecting. One of our initial reasons for adding this helper was to catch insecure behavior in `tarfile` and `zipfile`, and both of these libraries have a [common pattern of instantiation via the `with` statement](https://docs.python.org/3/library/tarfile.html#examples).

Let's add context manager variable instantiation support to `BadNameAttributeUseLinter`.
